reproject/utils.py

```diff
@@ -208,11 +209,19 @@ def parse_input_weights(input_weights, hdu_weights=None):
                 )
             else:
                 hdu_weights = 0
-        return parse_input_data(input_weights[hdu_weights])[0]
+        return parse_input_data(input_weights[hdu_weights])[slc]
     elif isinstance(input_weights, PrimaryHDU | ImageHDU | CompImageHDU):
-        return input_weights.data
+        if return_wcs:
+            ww = WCS(input_weights.header)
+            ww = ww if ww.has_celestial else None
```

Comment:
Here we again break compatibility with APE 14 WCS - why is it needed?

Reply:
This check is serving a critical role and cannot be removed, so we have to think of a replacement.

This check is asking whether the FITS header contains a WCS that should be interpreted as the WCS for the weights.  Since `WCS()` (i.e., `WCS` with an empty dict as input) is still a valid WCS object but contains no transforms, we need some check here whether the WCS exists.    Annoyingly, you can't just check that `naxis` matches `ndim`, since `naxis` defaults to 2.

Maybe...
```python
ww = ww if ww.array_shape == input_weights.data.shape
```
...I'll check this...

Reply:
no, I don't think that works because the shape comes from the data when reading.

Reply:
what is the APE14 way to check if there is a transformation associated with an axis?

Reply:
I don't think there's any general way to get at this.  All WCSes contain valid transforms, so the only solution is to use the full WCS validation.  I've added that now.  It's less clear for specific cases, but at least it solves the general case.